rs/sns/governance/proto/ic_sns_governance/pb/v1/governance.proto

```diff
@@ -2138,6 +2139,47 @@ message AdvanceTargetVersionRequest {
 // The response to a request to advance the target version of the SNS.
 message AdvanceTargetVersionResponse {}
 
+// Represents a single entry in the upgrade journal.
+message UpgradeJournalEntry {
+  oneof entry {
+    UpgradeStepsRefresh upgrade_steps_refresh = 1;
+    TargetVersionSet target_version_set = 2;
+    UpgradeStarted upgrade_started = 3;
+    UpgradeCompleted upgrade_completed = 4;
+  }
+  optional uint64 timestamp_seconds = 5;
+
+  message UpgradeStepsRefresh {
+    optional Governance.Versions upgrade_steps = 2;
```

Comment:
```suggestion
    repeated Governance.Version upgrade_steps = 1;
```

Reply:
Fails the candid compatibility checker if we want it to be optional in candid unfortunately